Resources/_template_week2_Haikus.md

```diff
@@ -0,0 +1,31 @@
+# Template for the haiku assignment
+
+## Instructions
+
+1. Form a subteam of three students from your team.
+2. As a subteam, create one branch on GitLab named `Group<groupnumber>/Haiku_<teamletter><subteamnumber>` (e.g. `Group1/Haiku_A1`).
+3. Copy this Template and rename it to the same name as the branch; keep the extension `.md` (e.g. `Group1/Haiku_A1.md`).
+4. One of the teammembers writes the first line of the haiku in the file on their local computer. Then uploads it to the branch.
+5. The next teammember pulls and edits the file, writing the second line of the haiku in the file. Then uploads it to the branch.
+6. The last teammember pulls and edits the file, and writing the third line of the haiku in the file. Then uploads it to the branch.
+7. One of the students creates a Merge Request to merge the file with the haiku into the master branch.
+8. The subteam assigns the other students from the team (but not from your subteam) to review the Merge Request.
+9. Once the reviewers have confirmed that it is a valid haiku, they can approve the Merge Request.
+10. One of the students from the subteam who wrote the haiku can merge the Merge Request.
```

Comment:
Last year we allowed the students to create, review, and approve merge requests, but not to actually perform the merge. This is to prevent accidents that can mess up the whole course repository (because it allows us to quickly check the pull request before merging it). Maybe an idea to keep this the same as last year?

Reply:
An idea could be to create three new repositories, for the Haikus of groups 1, 2 and 3; and let each subteam merge (and potentially break?) that repository. The main repository will then at least stay untouched. Is that an idea?

Reply:
Yes, I like that idea! That should also be feasible.

Reply:
> Yes, I like that idea! That should also be feasible.

OK, what do the other think? @SanliFaez, do you agree that I create three repositories for the three groups?

Reply:
I think that can be confusing. I restrict approval of merge requests to TAs, so all students will be developers and cannot merge to main.
therefore, I prefer to keep everything in just one repository, but restrict approving of merges to TAs.
if it is only for one exercise, you can also avoid a lot of conflicts by creating three sub-folders, one for each group. 

I think you still need to be more specific on where they need to add their subteam md files.

Reply:
okay, I have created a subfolder for this exercise under assignments. If students each create their Haiku under a separate filename as instructure, the merge requests will have no conflict with each other, and the three TAs can each merge the requests from their group without causing a conflict.